app/views/charges/_form.html.erb

```diff
@@ -0,0 +1,34 @@
+<%= form_with(model: charge) do |form| %>
+  <% if charge.errors.any? %>
+    <div style="color: red">
+      <h2><%= pluralize(charge.errors.count, "error") %> prohibited this charge from being saved:</h2>
+
+      <ul>
+        <% charge.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field">
+    <%= form.text_field :name, placeholder: "Name" %>
+  </div>
+
+  <div class="field">
+    <%= form.text_field :amount, placeholder: "Amount" %>
+  </div>
+
+  <div class="categorie-input-container">
+  <%= form.select :group_id, Group.all.collect {|group| [group.name, group.id]},{
+  include_blank: true} , class:"form-select" %>
+  </div>
+
+  <div class="add-t-container">
+    <%= form.submit 'SAVE', class: "add-transaction" %>
+  </div>
+
+  <div class="button-back">
+  <%= link_to "Back to charges", charges_path, class: "back-to" %>
+  </div>
+<% end %>
```

Comment:
- [ ] Nice job handling errors for the name and amount fields, I would also recommend you handle errors for the category field. When I try to add a transaction without a category I get this page.

  I think we can handle this better for a better experience by making the field compulsory and rendering error message just like the ones for name and amount.

  ![image](https://user-images.githubusercontent.com/36542986/209058094-1d8088e1-b171-4314-a7f9-fde6c96763e6.png)

Reply:
Thank you @TedLivist for your feeedback. I already implemented all the changes requested 👍🏼 .